packages/template-retail-react-app/app/components/_app/index.test.js

```diff
@@ -82,4 +85,68 @@ describe('App', () => {
         expect(hreflangLinks.some((link) => hasGeneralLocale(link))).toBe(true)
         expect(hreflangLinks.some((link) => link.hrefLang === 'x-default')).toBe(true)
     })
+
+    test('App component updates the basket with correct currency and customer email', async () => {
+        // Test basket. _app will be manipulating this basket for this test
+        const basket = {
+            basketId: 'asdfgh',
+            currency: 'CAD',
+            customerInfo: {
+                customerId: '12345'
+            }
+        }
+
+        jest.spyOn(
+            jest.requireActual('@salesforce/retail-react-app/app/hooks/use-current-customer'),
+            'useCurrentCustomer'
+        ).mockImplementation(() => {
+            return {
+                data: {
+                    customerId: '12345',
+                    email: 'email@test.com',
+                    isRegistered: true
+                }
+            }
+        })
+
+        jest.spyOn(
+            jest.requireActual('@salesforce/retail-react-app/app/hooks/use-current-basket'),
+            'useCurrentBasket'
+        ).mockImplementation(() => {
```

Comment:
I am probably just overlooking something but I couldn't get this to work with the following jest.mock (and similar for useCurrentCustomer) but it works with jest.spyOn so this looks like this.

```
jest.mock('@salesforce/retail-react-app/app/hooks/use-current-basket', () => ({
     useCurrentBasket: () => {
        return {
             data: basket,
             derivedData: {
                 hasBasket: true,
                 totalItems: 0
             }
         }
     }
```

Reply:
Looking at the top of the test class, `useMultiSite` is mocked with relative paths. Maybe that's why my attempts at mocking use-current-basket with `jest.mock` wasn't working?

Reply:
I got a variation of this working that applies just to the new test. Combined with some other changes, using `jest.mock` rather than `jest.spyOn` allows me to remove the `act` also.